python/cudf/tests/test_joining.py

```diff
@@ -335,3 +335,29 @@ def test_dataframe_empty_merge():
     got = gdf1.merge(gdf2, how='left', on=['a'])
 
     assert_eq(expect, got)
+
+
+def test_dataframe_merge_order():
```

Comment:
Please add a test for an empty merge as well because the order differs in that case for pandas.

Reply:
I'm planning on adding a bunch more tests to this before I finalize it. `empty_merge` test already exists immediately above.

Reply:
By bunch I mean an exhaustive set: numerical, categorical, datetime, inner, outer, left, right, square matrix, column major, row major, and the set of tuples of columns in `abc`.

Reply:
I wrote some datetime tests but it appears pandas does not support merging on datetimes, so I am unsure how to proceed. Otherwise this PR is finished.